server/src-lib/Hasura/Server/Logging.hs

```diff
@@ -192,7 +191,10 @@ data OperationLog
   { olRequestId          :: !RequestId
   , olUserVars           :: !(Maybe UserVars)
   , olResponseSize       :: !(Maybe Int64)
-  , olQueryExecutionTime :: !(Maybe Double)
+  , olQueryIoTime        :: !(Maybe NominalDiffTime)
+  -- ^ Request IO wait time, i.e. time spent reading the full request from the socket.
```

Comment:
Can the field name can be slightly more explicit (I feel that `query_io_time` suggests that this is the total time spent in io), maybe something like `request_read_time`?

Reply:
How hard would it be implement the entire request time (https://github.com/hasura/graphql-engine/issues/3547)?

Reply:
I agree that name is clearer

> How hard would it be implement the entire request time 

It seemed more useful to me to measure and report the network IO time directly, and you can add the two times together to get total (except for network IO time for sending the response) time. 

But if you are asking about how we can measure the network IO time for sending the response: this looked really difficult/invasive to do, since at the highest level we're passing spock continuations like `... -> Handler m (HttpResponse EncJSON)`. So it looked to me like we'd have to get into the internals of spock to do this, and it would be quite inconvenient. But I haven't looked into it deeply.

My thought was for the purposes of the underlying issue in #3547  (slow network issues being discoverable) measuring in one direction would be sufficient. Let me know what you think!

Reply:
> But if you are asking about how we can measure the network IO time for sending the response: this looked really difficult/invasive to do, since at the highest level we're passing spock continuations like ... -> Handler m (HttpResponse EncJSON). So it looked to me like we'd have to get into the internals of spock to do this, and it would be quite inconvenient. But I haven't looked into it deeply

Fair enough, `request_read_time` should do for now.